lib/discordrb/voice/network.rb

```diff
@@ -98,10 +98,21 @@ def send_audio(buf, sequence, time)
     # Sends the UDP discovery packet with the internally stored SSRC. Discord will send a reply afterwards which can
     # be received using {#receive_discovery_reply}
     def send_discovery
-      discovery_packet = [@ssrc].pack('N')
+      # Create empty packet
+      discovery_packet = ''
 
-      # Add 66 zeroes so the packet is 70 bytes long
+      # Add Type request (0x1 = request, 0x2 = response)
+      discovery_packet += [0x1].pack('n')
+
+      # Add Length (excluding Type and itself = 70)
+      discovery_packet += [70].pack('n')
+
+      # Add SSRC
+      discovery_packet += [@ssrc].pack('N')
+
+      # Add 66 zeroes so the packet is 74 bytes long
```

Comment:
More efficient way to handle it
```suggestion
     # request type, packet length and SSRC padded to 70 bytes
      discovery_packet = [0x1, 70, @ssrc].pack('nnNx66')
```

Reply:
I totally agree that it is more efficient. I had hesitated to do it this way, but then I thought that if Discord changed the package again, it would be easier for anyone to come back and add what was missing from the package by just looking at the Discord doc like I did.
If you want me to reduce the package building process to the bare minimum though, I can do that.

Reply:
We can leave it this way. More readable is probably better than minor efficiency anyway

Reply:
In theory the call to this function is only made at the connection in a voice room, unless you have a bot that spends its time doing this, there is little chance that non-optimization would have any negative impact.

Then I can see to compact the 3 first ones (type, length and ssrc) by commenting well at least, and by leaving the padding "\0" apart ?

Reply:
I think we should leave it exactly as you have it. If we switch to a complicated pack call we end up making code that relatively few people can change in the future. I was a bit over zealous in optimization 🙂